xarray/core/groupby.py

```diff
@@ -234,7 +234,45 @@ def __init__(self, obj, group, squeeze=False, grouper=None, bins=None,
                 # TODO: sort instead of raising an error
                 raise ValueError('index must be monotonic for resampling')
             s = pd.Series(np.arange(index.size), index)
-            first_items = s.groupby(grouper).first()
+            if isinstance(grouper, tuple):
```

Comment:
I've been thinking about this logic over the last few days.  With some minor modifications to the `_get_time_bins` logic, I think there may be a simpler way of determining this `first_items` Series without distinguishing between downsampling and upsampling.  I'll try to post a gist and some more comments later today or tomorrow.